scripts/globals/mobskills/severing_fang.lua

```diff
@@ -0,0 +1,38 @@
+-----------------------------------
+--  Severing Fang
+--
+--  Description: Bites at all targets in front. Additional effect: Defense Down
+--  Type: Physical
+--  Utsusemi/Blink absorb: 2 shadows
+--  Range: Front arc (cone)
+-----------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/monstertpmoves")
+-----------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+  -- Do not use this weapon skill on targets behind. Sub-Zero Smash
+  -- should trigger in this case.
+  if target:isBehind(mob) then
+    return 1
+  end
+  return 0
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local numhits = 2
+    local accmod = 1
+    local dmgmod = 1.5
+    local info = MobPhysicalMove(mob, target, skill, numhits, accmod, dmgmod, TP_NO_EFFECT)
+    local dmg = MobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.PIERCING, MOBPARAM_WIPE_SHADOWS)
+
+    local typeEffect = xi.effect.DEFENSE_DOWN
+    MobPhysicalStatusEffectMove(mob, target, skill, typeEffect, 30, 0, 60)
```

Comment:
I'm happy to do whatever the project prefers.

I'm a strong believer in placing local variables as close to their usages as possible. I believe it helps with readability (you don't have to jump earlier in the file to find out what value it holds) and it narrows the scope (i.e. lifetime) of the local.

But if the coding convention is to declare all locals ahead of the primary logic, I'm happy to move it up. Whatever is most consistent with the project.

Reply:
Honestly? If it makes sense, I'm on board with it.  That's a huge block of variables above, and I can see the reasoning here.

Reply:
I'll leave this open for now. I'm interested to see other folk's opinions.

(This may be another instance of my work-style-guide heavily influencing my coding style haha.)

Reply:
if its used multiple places in the script and or avoid a wide line by making it a var, or if the variable ever gets modified later in the code: its good its a variable.

In cases where "typeEffect" only is ever called once, the line is not too wide with it in there, and its never modified: my preference is not to put it into a variable in that case.

just my 2 cents.

Reply:
Sounds good. I'll inline it then, in this case. I'm not sure (by that description) that the variable buys much: the line length isn't an issue, and the variable name adds no additional clarity.

Reply:
As far as _placement_ of _where_ to put a variable.. I only really care if the placement changes the scope. Sensible scope, and I am in the habit of putting all vars that have the same scope together, There's no rule on that I am away of /shrug so w/e works bro :+1: